File: field.cpp
#include <iostream>
#include <stdlib.h>
#include "field.hpp"

namespace field_functions {

    Field convertFromScreenInput(const char c) {
        switch (c) {
            case '#':
                return Field::WALL;
            case '@':
                return Field::MAN;
            case '.':
                return Field::GOAL;
            case ' ':
                return Field::EMPTY;
            case '$':
                return Field::BLOCK;
            case '*':
                return Field::BLOCK_ON_GOAL;
            case '+':
                return Field::MAN_ON_GOAL;
            default:
                std::cerr << "Invalid character '" << c << "' in screen!" << std::endl;
                abort();
        }
    }

    std::string convertToSmvString(const Field field) {
        switch (field) {
            case Field::WALL:
                return "w";
            case Field::MAN:
                return "m";
            case Field::GOAL:
                return "g";
            case Field::EMPTY:
                return "e";
            case Field::BLOCK:
                return "b";
            case Field::BLOCK_ON_GOAL:
                return "bog";
            case Field::MAN_ON_GOAL:
                return "mog";
        }
    }
}

File: src/main/java/sokoban/Field.java
package sokoban;

public enum Field {

  EMPTY, WALL, MAN, BLOCK, GOAL, BLOCK_ON_GOAL, MAN_ON_GOAL;

  public static Field convertFromScreenInput(final char c) {
    switch (c) {
      case '#':
        return WALL;
      case '@':
        return MAN;
      case '.':
        return GOAL;
      case ' ':
        return EMPTY;
      case '$':
        return BLOCK;
      case '*':
        return BLOCK_ON_GOAL;
      case '+':
        return MAN_ON_GOAL;
      default:
        throw new IllegalArgumentException(String.format("Illegal character %s in screen!", c));
    }
  }

  public String convertToSmvString() {
    switch (this) {
      case WALL:
        return "w";
      case MAN:
        return "m";
      case GOAL:
        return "g";
      case EMPTY:
        return "e";
      case BLOCK:
        return "b";
      case BLOCK_ON_GOAL:
        return "bog";
      case MAN_ON_GOAL:
        return "mog";
      default:
        throw new IllegalArgumentException("Invalid Field " + this);
    }
  }
}
